lib/utils/addSemicolonForEditInfo.mjs

```diff
@@ -0,0 +1,30 @@
+/** @import { Node as PostcssNode } from 'postcss' */
+
+/**
+ * Adjust the text in EditInfo to include a semicolon when needed.
+ *
+ * @param {PostcssNode} node
+ * @param {{range: [number, number], text: string}} fixData
+ * @returns {{range: [number, number], text: string}}
+ */
+export default function addSemicolonForEditInfo(node, fixData) {
+	if (node.type === 'decl') {
+		if (node.parent?.raws.semicolon || node.parent?.last !== node) {
+			return {
+				...fixData,
+				text: `${fixData.text};`,
+			};
+		}
+	}
+
+	if (node.type === 'atrule') {
```

Comment:
note

[`@charset` is also considered an at-rule by postcss.](https://github.com/postcss/postcss/blob/main/lib/at-rule.d.ts#L71-L74)